mbed_tools_ci_scripts/utils/package_helpers.py

```diff
@@ -0,0 +1,178 @@
+"""Utilities for retrieving Python's package information."""
+
+import logging
+import re
+from typing import List, Any, Optional
+
+import pkg_resources
+
+from mbed_tools_ci_scripts.utils.configuration import ConfigurationVariable, configuration
+from .definitions import UNKNOWN
+
+logger = logging.getLogger(__name__)
+
+
+class PackageMetadata:
+    """Package's metadata.
+
+    Retrieves all the information it needs from Python's metadata dictionary.
+    It is based on https://www.python.org/dev/peps/pep-0314/
+    https://packaging.python.org/specifications/core-metadata/
+    """
+
+    def __init__(self, data: dict) -> None:
+        """Constructor."""
+        self._data = data
+
+    @property
+    def name(self) -> str:
+        """Gets package's name."""
+        return self._data.get('Name', UNKNOWN)
+
+    @property
+    def version(self) -> str:
+        """Gets package's version."""
+        return self._data.get('Version', UNKNOWN)
+
+    @property
+    def author(self) -> str:
+        """Gets package's author."""
+        return self._data.get('Author', UNKNOWN)
+
+    @property
+    def author_email(self) -> str:
+        """Gets package's author email."""
+        return self._data.get('Author-email', UNKNOWN)
+
+    @property
+    def licence(self) -> str:
+        """Gets package's licence."""
+        return self._data.get('License', UNKNOWN)
+
+    @property
+    def description(self) -> str:
+        """Gets package's licence."""
+        return self._data.get('Summary', UNKNOWN)
+
+    @property
+    def url(self) -> str:
+        """Gets package's URL."""
+        home_page = str(self._data.get('Home-page'))
+        if home_page:
+            return home_page
+        url = str(self._data.get('Project-URL'))
+        if url:
+            return url.split(',')[1].strip()
+        return UNKNOWN
+
+    def __str__(self) -> str:
+        """String representation."""
+        relevant_data = [
+            f'{getter}: {getattr(self, getter, None)}' for getter
+            in dir(self) if not getter.startswith("_") and getter
+        ]
+        return f'{", ".join(relevant_data)}'
+
+
+class ProjectMetadata:
+    """Metadata for a project."""
+
+    def __init__(self, package_name: str) -> None:
+        """Constructor."""
+        self._package_metadata: PackageMetadata = PackageMetadata(dict())
+        self._dependency_packages_metadata: List[PackageMetadata] = []
```

Comment:
Curiosity only: 
Why use `dict()` and `[]`, 
why not `{}` and `[]` 
or `dict()` and `list()`?